digiroad2-api-common/src/main/scala/fi/liikennevirasto/digiroad2/Digiroad2Context.scala

```diff
@@ -71,6 +71,25 @@ class ValluTerminalActor(massTransitStopService: MassTransitStopService) extends
   }
 }
 
+class ValluPublicStops(massTransitStopService: MassTransitStopService, publicStopService: MassTransitStopService) extends Actor {
+  val municipalityService: MunicipalityService = Digiroad2Context.municipalityService
+  def withDynSession[T](f: => T): T = publicStopService.withDynSession(f)
+  def receive = {
+    case publicStop: ServicePoint => persistedAssetChanges(publicStop)
+    case _      => println("received unknown message")
+  }
+
+  def persistedAssetChanges(publicStop: ServicePoint) = {
+    withDynSession {
+      val municipalityName = municipalityService.getMunicipalityNameByCode(publicStop.municipalityCode, false)
+      val massTransitStop = EventBusMassTransitStop(publicStop.municipalityCode, municipalityName, publicStop.nationalId,
+        publicStop.lon, publicStop.lat, None, None, publicStop.created, publicStop.modified, publicStop.propertyData)
+      ValluSender.postToVallu(massTransitStop)
+      massTransitStopService.saveIdPrintedOnValluLog(publicStop.id)
+    }
+  }
+}
+
```

Comment:
The new bus stop types will not be included in vallu-xml-messages so this can be deleted